src/main/java/com/carpool/car_pool/controllers/RideOfferController.java

```diff
@@ -36,19 +36,28 @@ public ResponseEntity<Long> createRideOffer(
         return ResponseEntity.ok(rideOfferService.createRideOffer(rideOfferRequest, currentUser));
     }
 
-    @GetMapping("/details")
+    @GetMapping("/details/{id}")
     public ResponseEntity<RideOfferResponse> viewRideOfferDetails(
-            @RequestParam @Valid Long ID
-            ) {
-        return ResponseEntity.ok(rideOfferService.viewRideOfferDetail(ID));
+            @PathVariable Long id
+    ) {
+        return ResponseEntity.ok(rideOfferService.viewRideOfferDetail(id));
     }
 
-    @PutMapping("/details")
+    @PutMapping("/details/{id}")
     public ResponseEntity<RideOfferResponse>  editRideOfferDetails(
-            @RequestBody EditRideOfferRequest editRideofferRequest
+            @RequestBody EditRideOfferRequest editRideOfferRequest,
+            @PathVariable String id // TODO: does this make sense? should I check if id == editRideOfferRequest.id?
     ){
         UserEntity currentUser = currentUserService.getCurrentUser();
-        return ResponseEntity.ok(rideOfferService.editRideOfferDetail(editRideofferRequest, currentUser));
+        return ResponseEntity.ok(rideOfferService.editRideOfferDetail(editRideOfferRequest, currentUser));
     }
 
+    @DeleteMapping("/details/{id}")
+    public ResponseEntity<Void> deleteRideOffer(
+            @PathVariable Long id
+    ) {
+        UserEntity currentUser = currentUserService.getCurrentUser();
+        rideOfferService.deleteRideOffer(id, currentUser);
+        return ResponseEntity.noContent().build();
```

Comment:
Delete in HTTP should return 200, 202, or 204 response on successful operation

Reply:
return ResponseEntity.ok().build(); does return 200 OK right?

Reply:
Yes that is correct